### PR TITLE
Send a reminder email for new versions

### LINKED ISSUE #592: Email notifications: "Depositor: New version in progress, not submitted/published"
https://docs.google.com/document/d/1TZVILUrORI25kRZyUZ70vMQYo90Ngp_tqCI7BdU_VyQ/edit#heading=h.awan2wkp6vwt

Example implementation: https://github.com/sul-dlss/happy-heron/pull/673

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

Fixes #592

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ RSpec/ExampleLength:
   Enabled: false
 
 RSpec/MultipleMemoizedHelpers:
-  Max: 15
+  Max: 14
```
Comment (mjgiarlo) [inline .rubocop.yml:49]: :man_dancing:

File: app/services/work_reminder_generator.rb [hunk 2]
```diff
@@ -5,29 +5,37 @@
...
-  sig { params(first_interval: Integer, subsequent_interval: Integer).returns(ActiveRecord::Relation) }
+  sig { params(first_interval: Integer, subsequent_interval: Integer).returns(T::Hash[Symbol, ActiveRecord::Relation]) }
   private_class_method def self.eligible_works(first_interval, subsequent_interval)
-    Work
-      .with_state(:first_draft)
-      .where('(((CURRENT_DATE - CAST(created_at AS DATE)) - :first_interval) % :subsequent_interval) = 0',
-             first_interval: first_interval, subsequent_interval: subsequent_interval)
+    %i[first_draft version_draft].index_with do |state|
+      Work.with_state(state)
+          .where('(((CURRENT_DATE - CAST(created_at AS DATE)) - :first_interval) % :subsequent_interval) = 0',
```
Comment (mjgiarlo) [inline app/services/work_reminder_generator.rb:37]: Can you add a comment that gives an example of these values so it's a little clearer what's going on here?

**Review by mjgiarlo — APPROVED** [review]
Approve pending consideration of comment suggestion

File: app/services/work_reminder_generator.rb (hunk 2 above)
Comment (jcoyne) [reply app/services/work_reminder_generator.rb:37]: @jmartin-sul actually came up with this approach, so maybe he could explain what's going on. I've just added a new state to the mix.

Comment (mjgiarlo) [reply app/services/work_reminder_generator.rb:37]: @jcoyne works for me if @jmartin-sul would consider a follow-up PR

**Pull request merged**